### DISCUSSION TITLE
m3u file open in editor not in the opener specify in config file.

### DISCUSSION THREAD

**Discussion by OriLiMu** [sxyazi/yazi#1224] [1. Q&A] [answered]
I want open `m3u` file in `mpv`

```bash
xdg-mime query filetype pl.m3u                                   ─╯
audio/x-mpegurl
```

I see that m3u is a audio file. 

 # `yazi.toml` 

```toml
play = [
	{ run = 'mpv "$@"', orphan = true, for = "unix" },
	{ run = 'mpv "%1"', orphan = true, for = "windows" },
	{ run = '''mediainfo "$1"; echo "Press enter to exit"; read''', block = true, desc = "Show media info", for = "unix" },
]

[open]
rules = [
	{ mime = "audio/*", use = [ "play", "reveal" ] },
```

But when I click <kbd>enter</kbd> on the m3u file. it open in editor (nvim). How to fix it?

**Comment by lpnh** [comment]
What is the output when you run `xdg-mime query default audio/x-mpegurl` ?

**Reply by OriLiMu** [reply]
```sh
 xdg-mime query default audio/x-mpegurl                          ─╯
org.kde.elisa.desktop
```

**Reply by lpnh** [reply]
Yeah, I have no idea where the problem could be. I can't reproduce it in my environment. The only thing I can suggest is to try Yazi nightly if you haven't already. Sorry.

**Comment by sxyazi** [comment]
You should use `file` instead of `xdg-mime`, https://yazi-rs.github.io/docs/configuration/yazi#open

**Reply by OriLiMu** [reply]
1. I use the **file** command  the mime type is `text/plain` anyway to change it? Make the `.m3u` file recognized as "video/*"
2. And I change the config to this still not work. 
```toml
[open]
open-m3u = [{ run = 'mpv "$@"', orphan = true, desc = "m3u" }]
append_rules = [
	{ name = "*.m3u", use = ["open-m3u", "reveal"] },
]
```

**Reply by sxyazi** [reply, marked as answer]
You've confused the difference between "open" and "opener":

- opener docs: https://yazi-rs.github.io/docs/configuration/yazi#opener
- open docs: https://yazi-rs.github.io/docs/configuration/yazi#open

**Reply by sxyazi** [reply]
Also confused the difference between `prepend_*` and `append_*`, see https://yazi-rs.github.io/docs/configuration/overview#mixing

**Reply by OriLiMu** [reply]
> Also confused the difference between `prepend_*` and `append_*`, see https://yazi-rs.github.io/docs/configuration/overview#mixing

In the doc

> And instead want to customize your keys upon the default, you can use prepend_* or append_* directories to prepend or append them to the default (See [keymap.toml](https://yazi-rs.github.io/docs/configuration/keymap) for details)

I know the difference between prepend and append, but the doc seems not said the effect of prepend / append. 

I guess that there is an order to get the match type/name from the rules (prepend_rules > rules > append_rules). If one file is match the prepend the check for rules will stop. Is that right?

**Reply by sxyazi** [reply]
Yep, Yazi will pick the first matching run.

> Since Yazi selects the first matching key to run, prepend always has a higher priority than default, and append always has a lower priority than default
